cmd/soroban-rpc/internal/test/simulate_transaction_test.go

```diff
@@ -500,7 +499,7 @@ func TestSimulateInvokeContractTransactionSucceeds(t *testing.T) {
 	require.Contains(t, metrics, "soroban_rpc_json_rpc_request_duration_seconds_count{endpoint=\"simulateTransaction\",status=\"ok\"} 3")
 	require.Contains(t, metrics, "soroban_rpc_preflight_pool_request_ledger_get_duration_seconds_count{status=\"ok\",type=\"db\"} 3")
 	require.Contains(t, metrics, "soroban_rpc_preflight_pool_request_ledger_get_duration_seconds_count{status=\"ok\",type=\"all\"} 3")
-	require.Contains(t, metrics, "soroban_rpc_preflight_pool_request_ledger_entries_fetched_sum 42")
+	require.Contains(t, metrics, "soroban_rpc_preflight_pool_request_ledger_entries_fetched_sum 48")
```

Comment:
why did this go up? :detective:

Reply:
Moar accesses :) 

The preflight algorithm changed. The same entries are queried more than once, but they are heavily cached now.

Reply:
Cool, just double-checking :) I didn't see any big changes to the test itself so I wanted to ensure this was explainable